### ISSUE TITLE
feature request: overheating values are not correct and no correct results

### ISSUE THREAD

**Issue by kOepi123** [acemod/ACE3#2317]
hello guys,

I heard there compatibility issues with the overheating on some mods like RHS, I had a quick
look at the content of the fnc_overheating and weaponcfgs.
I have to say that there are a lot of zeros in the values ;).

It looks like a quite complex function with many variables, which is, in my opinion not necessary.
The bulletmass has no impact on the overheating, the velocity varies only little to have a big impact, the barrelmass aswell has only little influence, because the barrels are all air-cooled and mostly meant to be transported by infantry, therefore lightweight design. The jam chance is irrelevant, as it only increases with uncovered belts that lay in the dust, not those that are protected by a drum magazine or similar.

There are only a few different profiles for the weaponry in the game.
Heavy machine guns, 12.7
Automatic grenade launcher, 40.0 = no overheating at all
Medium machine guns, 7.62
miniguns, multiple barrel machine gun, 7.62
intermediate machine guns, 6.5
light machine guns, 5.56
Assault rifles, 5.56
Assault rifles, 6.5
Assault rifle, 7.62
submachine guns, 9mm or .45

=> calibercoefficent * weapontypecoefficent * heatingcoefficent * volumeoffire
=> calibercoefficent * weapontypecoefficent* coolingcoefficent * timespan

so 10 different weapontype coefficent and 4 calibercoefficents.

----------------------------------------------------------------------------------------------
currently observed:
1. overheating is way too slow ( nevgev 7.62 after 800 rounds is still not "hot" )
2. cooldown is way too fast ( if I reload a little slower it is not overheating at all )
3. the only result of a hot weapon is a "jam"
4. complex overheating function prevents mods to be compatible ( I think a simpler structure would help, you know better if that is applicable )
----------------------------------------------------------------------------------------------
expected:
1. overheating working correctly at cyclic, sustained and rapid rates for machineguns and rifles
2. cooldown working correctly, spare barrel e.g. 10 minutes to be used again.
3. when a barrel is hot, the cartridges that are loaded will ignite ( cookoff ) and start a chain reaction ( runaway gun ) unitl no ammunition is fed anymore, while damaging the barrel and decreasing its lifespan heavily. I am aware of modern machineguns are technically designed to prevent cookoffs and runawaygun incidents, but it is still part of the weapon manuals and brings in another degree of variety.
- adding the cookoff for any weapon when it is hot and add runaway for automatic weapons or
weapons in the cyclic fire mode.
4. as I suggested above, a simpler function to reach all the modifications of the community.
------------------------------------------------------------------------------------------------

realistic rates of fire are stated in my calculation table which is uploaded here:

https://fileshack.net/7o652opztlvu

sources for my data: 
-FM 3-22.68 crew served machine guns
-TM 9-1005-249-10 M16 operator manual
-TM 9-11005-319-23&P M4 and m16 field maintenance manual
-FM 23-8 M14 rifle
-FM 23-65 M2 HB machine gun

**Comment by commy2** [comment]
>1. overheating is way too slow ( nevgev 7.62 after 800 rounds is still not "hot" )
>2. cooldown is way too fast ( if I reload a little slower it is not overheating at all )

I totally agree. Never had a single jam since release. (while actually playing the game)

**Comment by kOepi123** [comment]
updated calculation table with corrected definition of cooldown function :
https://fileshack.net/vmf8t53f5xz0

**Comment by PabstMirror** [comment]
I think PR #3082 should address some of these issues
>~1.333x increased heating
~1.45x less cooling

Overheated weapons will jam much more often and will have dispersion when hot
Configs have been simplified

Because of engine limitations, we really can't get a proper simulation of spare barrel cooling as they are simple stateless items right now.

**Comment by esteldunedain** [comment]
> Because of engine limitations, we really can't get a proper simulation of spare barrel cooling as they are simple stateless items right now.

I do have plans to support them eventually (making the magazines with ID)

**Comment by kOepi123** [comment]
While playing around with the ace values for dispersion and jam chance I realized, that it is really hard to find a correct value.
For weapon handling dispersion and jam chance change little when the weapon is hot, the most effect on the cone of fire will have recoil, muzzle climb anyhow and the jam chance can be set to ridicilously low like 2 shots only, but the sensitivity doesnt allow it really to set it to double the jam chance when being cold.

**Comment by kOepi123** [comment]
@esteldunedain
would love to help tweaking or in other way that I can.
I have accurate data on realistic values.
